src/Appwrite/Platform/Workers/Webhooks.php

```diff
@@ -113,7 +120,19 @@ private function execute(array $events, string $payload, Document $webhook, Docu
         }
 
         if (false === \curl_exec($ch)) {
-            $this->errors[] = \curl_error($ch) . ' in events ' . implode(', ', $events) . ' for webhook ' . $webhook->getAttribute('name');
+            $errorCount = $webhook->getAttribute('errors', 0) + 1;
+            $lastErrorLogs = \curl_error($ch) . ' in events ' . implode(', ', $events) . ' for webhook ' . $webhook->getAttribute('name');
+            $webhook->setAttribute('errors', $errorCount);
+            $webhook->setAttribute('logs', $lastErrorLogs);
+
+            if ($errorCount > 9) {
+                $webhook->setAttribute('status', false);
+            }
+
+            $dbForConsole->updateDocument('webhooks', $webhook->getId(), $webhook);
+            $dbForConsole->deleteCachedDocument('projects', $project->getId());
+
+            $this->errors[] = $lastErrorLogs;
```

Comment:
Can we please double check what scenario this errors array is for? It scares me to think about concurrency with this variable, and also I cant see where we clear it to empty array.